### PR TITLE
Fix two indentation problems

### PR DESCRIPTION
**Pull request by jbrobst**
### 1
Accessing property names which have the same name as keywords that belong on the stack adds these keyword tokens to the stack and possibly increases indentation. The stack/indentation level is not accordingly decremented since the property access is not actually a block and doesn't have the matching end-token. For example,
```js
foo.a=1.1;foo.switch=2.2;foo.b=3.3;
```
becomes
```js
foo.a = 1.1;
foo.switch = 2.2;
  foo.b = 3.3;
```
I fixed this by changing `token.type` to `acorn.tokTypes.name` when a token with a keyword type comes after a dot token.

### 2 (Edit: fixed by #23)
The if-statement responsible for popping `switch` from the stack compared `token` itself to `}` rather than `token.type.label` (`ttl`). This could result in incorrect indentation in some cases, such as
```js
var foo=[function(){switch(1){}}];
```
becoming
```js
var foo = [
  function () {
    switch (1) {
    }
  }
  ];

```

### CODE REVIEW

**Comment by mozbugzi2** [comment]
The switch bug is already fixed in #23.

**Comment by jbrobst** [comment]
@darkwing I have rebased this PR on master and dropped the commit that fixed the same issue as #23  (2 in my original post).

I think this and #23 getting merged warrant a version bump to 0.2.4. I often need the Firefox debugger for debugging very large minified files, most of which are not prettified correctly without all of these changes.

File: pretty-fast.js [hunk 1]
```diff
@@ -845,6 +845,12 @@
       }
 
       ttk = token.type.keyword;
+
+      if (ttk && lastToken && lastToken.type.label == ".") {
+        token.type = acorn.tokTypes.name;
+        ttk = token.type.keyword;
```
Comment (jasonLaster) [inline pretty-fast.js]: why is this line needed?

Comment (jasonLaster) [reply pretty-fast.js]: just removed it because it is likely redundant

Comment (jbrobst) [reply pretty-fast.js]: I believe it actually is needed because `token.type` is reassigned, so `token.type.keyword` has changed and thus `ttk` needs to be updated.

Comment (jasonLaster) [reply pretty-fast.js]: hmm, i see what you're saying. Could you file a new PR with that change and a test that demonstrates it.

Comment (jbrobst) [reply pretty-fast.js]: `ttk` is only used once on line 867, which will never be reached in the case of reassigning `token.type` because `belongsOnStack` doesn't return true for the `name` token type. So currently there is no code path that is bugged and there's no test I could write for it. However, it's still an issue because in the future someone might use `ttk` and a bug will be introduced because it has an incorrect value in some cases. Should I just submit the PR without a test?

Comment (jasonLaster) [reply pretty-fast.js]: sure
